plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-paypal-buttons-controller.php

```diff
@@ -125,4 +135,30 @@ public function create_order( WP_REST_Request $request ) {
 			200
 		);
 	}
-}
+
+	/**
+	 * Cancel a PayPal payment. This is used to move the woocommerce order back to a draft status.
+	 *
+	 * @param WP_REST_Request $request The request object.
+	 * @return WP_REST_Response The response object.
+	 */
+	public function cancel_payment( WP_REST_Request $request ) {
+		$data = $request->get_json_params();
+
+		if ( ! is_array( $data ) || empty( $data['order_id'] ) ) {
+			return new WP_REST_Response( array( 'error' => 'Invalid request' ), 400 );
+		}
+
+		$order_id = $data['order_id'];
+		$order    = wc_get_order( $order_id );
+
+		if ( ! $order ) {
+			return new WP_REST_Response( array( 'error' => 'Order not found' ), 404 );
+		}
+
+		$order->update_status( OrderStatus::CHECKOUT_DRAFT );
```

Comment:
Curious why we are specifying `checkout-draft` vs just `draft` here.

Reply:
When a draft order is created with GET `wc/store/v1/checkout`, the order status is `checkout-draft`. I am reverting it to the same status.